CMakeLists.txt

```diff
@@ -0,0 +1,934 @@
+	log-tree.c ls-refs.c mailinfo.c mailmap.c match-trees.c mem-pool.c merge.c merge-blobs.c 
+	merge-recursive.c mergesort.c midx.c name-hash.c negotiator/default.c 
+	negotiator/skipping.c notes.c notes-cache.c notes-merge.c notes-utils.c object.c oidmap.c 
+	oidset.c packfile.c pack-bitmap.c pack-bitmap-write.c pack-check.c pack-objects.c 
+	pack-revindex.c pack-write.c pager.c parse-options.c parse-options-cb.c patch-delta.c 
+	patch-ids.c path.c pathspec.c pkt-line.c preload-index.c pretty.c prio-queue.c progress.c 
+	promisor-remote.c prompt.c protocol.c quote.c range-diff.c reachable.c read-cache.c rebase.c
+	rebase-interactive.c reflog-walk.c refs.c refs/files-backend.c refs/iterator.c 
+	refs/packed-backend.c refs/ref-cache.c refspec.c ref-filter.c remote.c replace-object.c 
+	repo-settings.c repository.c rerere.c resolve-undo.c revision.c run-command.c 
+	send-pack.c sequencer.c serve.c server-info.c setup.c sha1-array.c sha1-lookup.c 
+	sha1-file.c sha1-name.c shallow.c sideband.c sigchain.c split-index.c 
+	stable-qsort.c strbuf.c streaming.c string-list.c submodule.c submodule-config.c 
+	sub-process.c symlinks.c tag.c tempfile.c thread-utils.c tmp-objdir.c 
+	trace.c trace2.c trace2/tr2_cfg.c trace2/tr2_cmd_name.c trace2/tr2_dst.c 
+	trace2/tr2_sid.c trace2/tr2_sysenv.c trace2/tr2_tbuf.c trace2/tr2_tgt_event.c 
+	trace2/tr2_tgt_normal.c trace2/tr2_tgt_perf.c trace2/tr2_tls.c trailer.c transport.c 
+	transport-helper.c tree-diff.c tree.c tree-walk.c unpack-trees.c upload-pack.c url.c 
+	urlmatch.c usage.c userdiff.c utf8.c varint.c version.c versioncmp.c walker.c wildmatch.c 
+	worktree.c wrapper.c write-or-die.c ws.c wt-status.c xdiff-interface.c 
+	zlib.c) 
+
+add_library(libgit ${libgit_SOURCES} ${compat_SOURCES})
+#add_dependencies(libgit command-list)
+set(libxdiff_SOURCES
+	xdiff/xdiffi.c xdiff/xprepare.c xdiff/xutils.c xdiff/xemit.c 
+	xdiff/xmerge.c xdiff/xpatience.c xdiff/xhistogram.c)
+add_library(xdiff STATIC ${libxdiff_SOURCES})
+
+set(libvcs-svn_SOURCES
+vcs-svn/line_buffer.c vcs-svn/sliding_window.c vcs-svn/fast_export.c 
+vcs-svn/svndiff.c vcs-svn/svndump.c)
+add_library(vcs-svn STATIC ${libvcs-svn_SOURCES})
+
+#add git.rc
+if(WIN32)
+	add_custom_command(OUTPUT ${CMAKE_BINARY_DIR}/git.res
+			COMMAND ${WINDRES_EXE} -O coff -DMAJOR=${PROJECT_VERSION_MAJOR} -DMINOR=${PROJECT_VERSION_MINOR}
+				-DMICRO=0 -DPATCHLEVEL=0 -DGIT_VERSION="\\\"${PROJECT_VERSION}.GIT\\\""
+				-i ${CMAKE_SOURCE_DIR}/git.rc -o ${CMAKE_BINARY_DIR}/git.res
+			WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+			VERBATIM)
+	add_custom_target(git-rc DEPENDS ${CMAKE_BINARY_DIR}/git.res)
+endif()
+
+#link all required libraries to common-main 
+add_library(common-main OBJECT common-main.c)
+
+target_link_libraries(common-main libgit xdiff ${ZLIB_LIBRARIES})
+if(Intl_FOUND)
+	target_link_libraries(common-main ${Intl_LIBRARIES})
+endif()
+if(Iconv_FOUND)
+	target_link_libraries(common-main ${Iconv_LIBRARIES})
+endif()
+if(WIN32)
+	
```

Comment:
Maybe the empty line wants to come before the `if(WIN32)` instead of after it?

Reply:
Sorry, did not notice it, it was fixed in the later commits